osbs/utils/__init__.py

```diff
@@ -588,6 +590,13 @@ def retry(*args, **kwargs):
     return do_retry_on_exception
 
 
+def user_warning_log_handler(self, msg, *args, **kwargs):
+    if isinstance(msg, dict):
```

Comment:
Can we have an abstraction instead of raw dict. Fucntion:

```
def log_user_warning(msg, ...potential_future_extensions):
    jsonized_msg = magic(....)
    logger.user_warning(jsonized_msg)
```

Reply:
You probably misunderstood what I wanted. I'd like to have and extra function, that will create proper json dict. Instead of calling
```
logger.user_warning({'msg': 'yolo'})
```
I want to have abstraction
```
logger.user_warning(msg='yolo')
```

Reply:
extensible for future:
```
logger.user_warning(msg='yolo', doc_ecode=123, kind='deprecation')
```

Reply:
User warning istself should be responsible for creating that dictionary used for json